src/conn/conn_api.c

```diff
@@ -2266,37 +2268,79 @@ __wt_verbose_config(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig)
         return (0);
     WT_RET(ret);
 
+    /* check if config "all" verbose */
```

Comment:
I wrote some pseudo-code, what do you think of the following approach to reduce the code changes:

```
diff --git a/src/conn/conn_api.c b/src/conn/conn_api.c
index 3e9b9ae6a..241fc9ddb 100644
--- a/src/conn/conn_api.c
+++ b/src/conn/conn_api.c
@@ -2267,7 +2267,13 @@ __wt_verbose_config(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig)
     WT_RET(ret);
 
     WT_RET(__wt_config_gets(session, cfg, "verbose", &cval));
+    /*
+     * Get the value for the keyword 'all'.
+     * If we get WT_NOTFOUND, set value_all to WT_VERBOSE_NOTICE
+     * If found, set value_all to the value associated with the keyword.
+     */
     for (ft = verbtypes; ft->name != NULL; ft++) {
+        /* If ft->name is 'all', go to next iteration. */
         ret = __wt_config_subgets(session, &cval, ft->name, &sval);
         WT_RET_NOTFOUND_OK(ret);
 
@@ -2277,7 +2283,7 @@ __wt_verbose_config(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig)
              * WT_VERBOSE_NOTICE verbosity level. WT_VERBOSE_NOTICE being an always-on informational
              * verbosity message.
              */
-            conn->verbose[ft->flag] = WT_VERBOSE_NOTICE;
+            conn->verbose[ft->flag] = value_all;
         else if (sval.type == WT_CONFIG_ITEM_BOOL && sval.len == 0)
             /*
              * If no value is associated with the event (i.e passing verbose=[checkpoint]), default
```
@y123456yz, @ajmorton, thoughts?

Reply:
Looks good to me!
It's a bit tricky to mentally apply this diff on top of the current changes. To confirm we're saying the same thing, the final changes in this PR will be:
- Have a new local variable called something like `value_all`/`default_verbose_level` which is initially set to `WT_VERBOSE_NOTICE`
- We first search for an `all` config, and if it's present set the value of `value_all` accordingly
- Then in the existing `for (ft = verbtypes; ft->name != NULL; ft++)` loop we'll skip the `all` category, and when a config item isn't found (`if (ret == WT_NOTFOUND)`) we'll set the category to `value_all` instead of the current `WT_VERBOSE_NOTICE`
?

Reply:
Yes, exactly!